executor/tiflash_test.go

```diff
@@ -139,3 +139,56 @@ func (s *tiflashTestSuite) TestMppExecution(c *C) {
 	// test proj and selection
 	tk.MustQuery("select count(*) from (select a * 2 as a from t1) t1 , (select b + 4 as a from t)t where t1.a = t.a").Check(testkit.Rows("3"))
 }
+
+func (s *tiflashTestSuite) TestPartitionTable(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("drop table if exists t1")
+	tk.MustExec("drop table if exists t2")
+	tk.MustExec("create table t(a int not null primary key, b int not null) partition by hash(a+1) partitions 4")
+	tk.MustExec("alter table t set tiflash replica 1")
+	tb := testGetTableByName(c, tk.Se, "test", "t")
+	err := domain.GetDomain(tk.Se).DDL().UpdateTableReplicaInfo(tk.Se, tb.Meta().ID, true)
+	c.Assert(err, IsNil)
+	tk.MustExec("insert into t values(1,0)")
+	tk.MustExec("insert into t values(2,0)")
+	tk.MustExec("insert into t values(3,0)")
+	tk.MustExec("insert into t values(4,0)")
+
+	tk.MustExec("create table t1(a int not null primary key, b int not null) partition by hash(a) partitions 4")
+	tk.MustExec("alter table t1 set tiflash replica 1")
+	tb = testGetTableByName(c, tk.Se, "test", "t1")
+	err = domain.GetDomain(tk.Se).DDL().UpdateTableReplicaInfo(tk.Se, tb.Meta().ID, true)
+	c.Assert(err, IsNil)
+	tk.MustExec("insert into t1 values(1,4)")
+	tk.MustExec("insert into t1 values(2,3)")
+	tk.MustExec("insert into t1 values(3,2)")
+	tk.MustExec("insert into t1 values(4,1)")
+
+	tk.MustExec("set @@session.tidb_isolation_read_engines=\"tiflash\"")
+	tk.MustExec("set @@session.tidb_allow_mpp=ON")
+	tk.MustExec("set @@session.tidb_opt_broadcast_join=ON")
+	// test if it is stable.
+	tk.MustQuery("select count(*) from t1 , t where t1.a = t.a").Check(testkit.Rows("4"))
+	// test partition prune
+	tk.MustQuery("select count(*) from t1 , t where t1.a = t.a and t1.a < 2 and t.a < 2").Check(testkit.Rows("1"))
+	tk.MustQuery("select count(*) from t1 , t where t1.a = t.a and t1.a < -1 and t.a < 2").Check(testkit.Rows("0"))
+	// test multi-way join
+	tk.MustExec("create table t2(a int not null primary key, b int not null) partition by hash(a*2) partitions 4")
+	tk.MustExec("alter table t2 set tiflash replica 1")
+	tb = testGetTableByName(c, tk.Se, "test", "t2")
+	err = domain.GetDomain(tk.Se).DDL().UpdateTableReplicaInfo(tk.Se, tb.Meta().ID, true)
+	c.Assert(err, IsNil)
+
+	tk.MustExec("insert into t2 values(1,0)")
+	tk.MustExec("insert into t2 values(2,0)")
+	tk.MustExec("insert into t2 values(3,0)")
+	tk.MustExec("insert into t2 values(4,0)")
+	tk.MustQuery("select count(*) from t1 , t, t2 where t1.a = t.a and t2.a = t.a").Check(testkit.Rows("4"))
+
+	// test avg
+	tk.MustQuery("select avg(t1.a) from t1 , t where t1.a = t.a").Check(testkit.Rows("2.5000"))
```

Comment:
should we ensure all query plans run in mpp?

Reply:
Good question, I will test the plans as well